sound/soc/sof/intel/hda-loader.c

```diff
@@ -470,3 +471,122 @@ int hda_dsp_post_fw_run(struct snd_sof_dev *sdev)
 	/* re-enable clock gating and power gating */
 	return hda_dsp_ctrl_clock_power_gating(sdev, true);
 }
+
+static int hda_dsp_ext_man_get_cavs_config_data(struct snd_sof_dev *sdev,
+				   const struct sof_ext_man_elem_header *hdr)
+{
+	const struct sof_ext_man_cavs_config_data *config_data =
+		container_of(hdr, struct sof_ext_man_cavs_config_data, hdr);
+	struct sof_intel_hda_dev *hda = sdev->pdata->hw_pdata;
+	int i;
+
+	for (i = 0; i < SOF_EXT_MAN_CAVS_CONFIG_LAST_ELEM - 1; i++)
+		switch (config_data->elems[i].token) {
+		case SOF_EXT_MAN_CAVS_CONFIG_CAVS_LPRO:
+			hda->clk_config = config_data->elems[i].value;
+			dev_dbg(sdev->dev, "FW clock config: %s\n", hda->clk_config ? "LPRO" : "HPRO");
+			break;
+		default:
+			dev_warn(sdev->dev, "unsupported token type: %d\n",
+				 config_data->elems[i].token);
+			break;
+		}
+
+	return 0;
+}
+
+// Fred: DUPLICATE warning, should we share?
```

Comment:
yes, please

Reply:
what is the duplicat warning? There are 2 separate things isn it? 1 we have an upsuppprted item in the platform config data, 2 we have an unsupported header in the ext manifest. How can we share?